### PR TITLE
Enhance releases API docs for CI_JOB_TOKEN usage

### LINKED ISSUE #1850: Allow to access Releases API via CI_JOB_TOKEN
## Description of the problem, including code/CLI snippet

Thanks to #746 we have support for the releases.
Unfortunately we have to load a project to create a release, but getting a project (event its own project like #1393 reported it) is not possible with CI_JOB_TOKEN.

I have made a work-around for this, by creating a fake Project object, but it's not very nice. Maybe we could have a way to use the Releases API with CI_JOB_TOKEN:

```
# fails with CI_JOB_TOKEN
gitlab_project = client.projects.get(id=1)
# So this work as we aren't querying
gitlab_project = client._objects.Project(client.projects, attrs={"id": 1})
# To be able to do
release = gitlab_project.releases.create({"name": "Supa name"})
```

## Expected Behavior

Not sure... Maybe something like this?

```
release = client.releases.create({"project_id": 1, "name": "Supa name"})
```

## Actual Behavior

See description above

## Specifications

  - python-gitlab version: ALL until at least 3.1
  - API version you are using (v3/v4): v4
  - Gitlab server version (or gitlab.com): 13/14.x

### PR DESCRIPTION
**Pull request by MRigal**
Fixes #1850 
@max-wittig @nejch willing to have a look on it?

### CODE REVIEW

File: docs/gl_objects/releases.rst [hunk 1]
```diff
@@ -45,6 +46,14 @@ Delete a release::
     # delete object directly
     release.delete()
 
+.. note::
+
+    The Releases API is one of the few working with CI_JOB_TOKEN, but the project can't
```
Comment (nejch) [inline docs/gl_objects/releases.rst:51]: nitpick: we usually wrap these variables like code I think
```suggestion
    The Releases API is one of the few working with ``CI_JOB_TOKEN``, but the project can't
```

File: docs/gl_objects/releases.rst [hunk 2]
```diff
@@ -45,6 +46,14 @@ Delete a release::
...
+    be fetched with the token. Thus use `lazy` for the project as in the above example.
+
+    Also be aware that most of the capabilities of the endpoint were not accessible per
+    CI_JOB_TOKEN until Gitlab version 14.5.
```
Comment (nejch) [inline docs/gl_objects/releases.rst:55]:
```suggestion
    Also be aware that most of the capabilities of the endpoint were not accessible with
    ``CI_JOB_TOKEN`` until Gitlab version 14.5.
```

**Review by nejch — CHANGES_REQUESTED** [review]
Thanks @MRigal for following up on this! Just a small nitpick, and I think commitlint might complain because of the capitilized message, so a little `--amend` might be needed :)

**Comment by MRigal** [comment]
Thanks @nejch for the review. My bad for the commit format, I had seen the pre-commit configs, but forgot to install the specific hook type, although precisely described here: https://github.com/python-gitlab/python-gitlab/blob/main/CONTRIBUTING.rst#development-workflow

Hope this fits the expectation now and hope this can help someone else :-)

**Review by nejch — APPROVED** [review]

**Comment by nejch** [comment]
Looks great, thanks again @MRigal! If you find anything else that doesn't make sense feel free to contribute as well! :)

**Pull request merged**